control-plane/agents/src/bin/core/node/wrapper.rs

```diff
@@ -1338,6 +1338,15 @@ impl ReplicaApi for Arc<tokio::sync::RwLock<NodeWrapper>> {
         }
     }
 
+    /// Resize an existing replica to the requested size, via gRPC.
+    async fn resize_replica(&self, request: &ResizeReplica) -> Result<Replica, SvcError> {
+        let dataplane = self.grpc_client_locked(request.id()).await?;
+        let replica = dataplane.resize_replica(request).await?;
+        let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
+        self.update_replica_states(ctx.deref_mut()).await?;
```

Comment:
This is not not necessary, only need to add replica itself

Reply:
ok. `share` and `unshare` also perhaps don't need this but have it redundantly.

Reply:
As mentioned, please add the replica...
I think share and unshare do it this way because they're an older api which doesn't return the replica but the share uri

Reply:
> As mentioned, please add the replica... I think share and unshare do it this way because they're an older api which doesn't return the replica but the share uri

Done, PTAL.